generator/src/main/java/com/scottlogic/deg/generator/generation/VelocityMonitor.java

```diff
@@ -0,0 +1,97 @@
+package com.scottlogic.deg.generator.generation;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
+import com.scottlogic.deg.generator.fieldspecs.RowSpec;
+import com.scottlogic.deg.generator.outputs.GeneratedObject;
+import com.scottlogic.deg.generator.walker.reductive.FixedField;
+import com.scottlogic.deg.generator.walker.reductive.ReductiveState;
+
+import java.math.BigInteger;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class VelocityMonitor implements ReductiveDataGeneratorMonitor {
+    private String startedGenerating;
+    private long rowsSinceLastSample;
+    private BigInteger rowsEmitted;
+    private BigInteger maxRows;
+    private Timer timer = new Timer(true);
+    private SimpleDateFormat simpleDateFormat = new SimpleDateFormat("hh:mm:ss");
+    private long previousVelocity = 0;
+
+    @Override
+    public void generationStarting(GenerationConfig generationConfig) {
+        this.startedGenerating = simpleDateFormat.format( new Date());
+        this.rowsSinceLastSample = 0;
+        this.rowsEmitted = BigInteger.ZERO;
+        this.maxRows = BigInteger.valueOf(generationConfig.getMaxRows());
+
+        System.out.println("Generation started at: " + this.startedGenerating + "\n");
+        System.out.println("Number of rows | Velocity (rows/sec) | Velocity trend");
+        System.out.println("---------------+---------------------+---------------");
+    }
+
+    @Override
+    public void rowEmitted(GeneratedObject row) {
+        this.rowsSinceLastSample++;
+        this.rowsEmitted = rowsEmitted.add(BigInteger.ONE);
+    }
+
+    @Override
+    public void endGeneration() {
+        reportVelocity(0);
+        System.out.println("\n\nGeneration finished at: " + simpleDateFormat.format(new Date()));
+    }
+
+    @Override
+    public void reportVelocity(long rowsSinceLastSample) {
+        String trend = rowsSinceLastSample > previousVelocity ? "+" : "-";
+        System.out.print(
+        String.format(
+            "%-14s | %-19d | %s \n",
+            this.rowsEmitted.toString(),
+            rowsSinceLastSample,
+            trend)
+        );
+        previousVelocity = rowsSinceLastSample;
+    }
+
+    public void startTimer() {
```

Comment:
There's no override on this method - does it come from the interface or not? On the class above you've had to override an empty implementation of this class?